### PR TITLE
Overhaul with ES6 and flow, bringing back up to date

### PR DESCRIPTION
**Pull request by ptomasroos**
This updated the code base to ES6.
Makes sure to move examples into expo in order to get reproduction repo's when people open new issues
I also moved away from eslint to prettier and made a commit hook in order to get consistent code styles, it was a major problem looking through the old PR's

### CODE REVIEW

**Comment by dsznajder** [comment]
@ptomasroos Finally someone took care about this great forgotten repo :D

**Comment by ptomasroos** [comment]
@dsznajder will do by best :)

**Comment by dsznajder** [comment]
After this PR I will also look into some opportunity to contribute as I forked this one a long time ago :)

**Comment by ptomasroos** [comment]
@dsznajder Perfect! Will make sure to keep PR's and complete issues up to date from now on.

**Comment by fforres** [comment]
Dang! This is a really cool PR

Kudos for reviving this project @ptomasroos 🤘

**Comment by malonguwa** [comment]
@ptomasroos hi man, thanks for looking after the project, I try to use your branch like this 
"react-native-scrollable-tab-view": "ptomasroos/react-native-scrollable-tab-view#overhaul",
in my RN 0.57.1 project. 
But after npm install, I got Can not find variable 'React' error when run the project. Not sure why is happens.

**Comment by ptomasroos** [comment]
Hi! Yes I’ve got some local commits to push will continue today!

**Comment by ptomasroos** [comment]
I will do the final refactorings and tests during tomorrow then we should hopefully be able to merge this later this week after someone more tried out the new branch

**Comment by dsznajder** [comment]
@ptomasroos When you think you will be able to merge this to master and release new version? :)

**Comment by ptomasroos** [comment]
@dsznajder late next week is my feeling, there are some major changes and I'm updating the examples along it to make issue tracking and maintainability easier, and since there aren't really any user facing improvements more from a maintainability and pr point of view I dont really see a reason to rush this. But doing my best to make it happen sooon!

**Comment by malonguwa** [comment]
@ptomasroos hi man, is this PR ready? Are there any more changes gonna made? Thanks again~

**Comment by ptomasroos** [comment]
Hi! Not ready, will continue to work with this during the week. And provide a better example repo. Will merge ASAP, but it still needs to be ready and working fully :)

**Comment by sin2** [comment]
Currently on expo sdk 31 (which uses RN 0.57) this module has issues compiling because of a trailing comma. This branch seems to fix this issue.

@ptomasroos 👍

**Comment by ptomasroos** [comment]
I would not recommend to use this branch @sin2 !
The latest release has the fixes with the trailing commas already published to npm

File: Button.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+type Props = React.ElementConfig<typeof TouchableOpacity> & {
+  children?: React.Node,
+}; // TODO React.ElementProps<typeof TouchableNativeFeedback>
+
+class Button extends React.Component<Props> {
+  renderAndroid() {
+    const { children, ...props } = this.props;
+
+    return (
+      <TouchableNativeFeedback
```
Comment (dsznajder) [inline Button.js]: `TouchableNativeFeedback` does not have `style` prop. If you passing a styles to this component you should wrap `View` with `TouchableNativeFeedback`

https://facebook.github.io/react-native/docs/touchablenativefeedback

File: package.json [hunk 2]
```diff
@@ -19,22 +18,34 @@
...
+      "git add"
+    ]
   },
-  "homepage": "https://github.com/brentvatne/react-native-scrollable-tab-view#readme",
+  "license": "MIT",
   "dependencies": {
-    "react-timer-mixin": "^0.13.3",
-    "prop-types": "^15.6.0",
-    "create-react-class": "^15.6.2"
+    "underscore": "^1.9.1"
```
Comment (dsznajder) [inline package.json]: Unused dependency :)

**Comment by malonguwa** [comment]
@ptomasroos Hi mate~ not try to push you~just want to ask is there any update on this PR~ looking forward to the updated version!

**Comment by malonguwa** [comment]
@ptomasroos happy new year~ any news for this PR ? :)

**Comment by andyr6381** [comment]
any update?

**Comment by malonguwa** [comment]
@ptomasroos 

I saw there is a breaking change related to the scrollView in React Native 0.58.x release.
Does this affect this lib if we upgrade to the 0.58.x ?

https://github.com/react-native-community/react-native-releases/blob/master/CHANGELOG.md#breaking-changes
 
Breaking Changes:
Public methods of components converted to ES6 classes are no longer bound to their component instance. For ScrollView, the affected methods are setNativeProps, getScrollResponder, getScrollableNode, getInnerViewNode, scrollTo, scrollToEnd, scrollWithoutAnimationTo, and flashScrollIndicators. For CameraRollView, the affected methods are: rendererChanged. For SwipeableRow, the affected methods are: close. Therefore, it is no longer safe to pass these method by reference as callbacks to functions. Auto-binding methods to component instances was a behaviour of createReactClass that we decided to not preserve when switching over to ES6 classes.

**Comment by ptomasroos** [comment]
@malonguwa we run on top of 0.58.7 for our releases and breakage in our apps regarding that

**Comment by malonguwa** [comment]
@ptomasroos thanks for the info, is there any eta on this PR to also address the breaking change issue?

**Comment by edo1493** [comment]
Hey guys, I have updated to "react-native": "^0.59.8" and all the sudden I have started seeing crashes in production, such as "TypeError: t.find is not a function." from this component.

I think this is caused by what @malonguwa mentioned.

**Pull request closed without merging**